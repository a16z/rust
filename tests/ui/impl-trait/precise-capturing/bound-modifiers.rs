//@ edition: 2021

#![feature(precise_capturing)]

fn polarity() -> impl Sized + ?use<> {}
//~^ ERROR expected identifier, found keyword `use`
//~| ERROR cannot find trait `r#use` in this scope
//~| WARN relaxing a default bound only does something for `?Sized`
//~| WARN relaxing a default bound only does something for `?Sized`

fn asyncness() -> impl Sized + async use<> {}
//~^ ERROR expected identifier, found keyword `use`
//~| ERROR cannot find trait `r#use` in this scope
//~| ERROR async closures are unstable

fn constness() -> impl Sized + const use<> {}
//~^ ERROR expected identifier, found keyword `use`
//~| ERROR cannot find trait `r#use` in this scope
//~| ERROR const trait impls are experimental

fn binder() -> impl Sized + for<'a> use<> {}
//~^ ERROR expected identifier, found keyword `use`
//~| ERROR cannot find trait `r#use` in this scope

fn main() {}
