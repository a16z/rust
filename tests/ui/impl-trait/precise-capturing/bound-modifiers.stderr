error: expected identifier, found keyword `use`
  --> $DIR/bound-modifiers.rs:5:32
   |
LL | fn polarity() -> impl Sized + ?use<> {}
   |                                ^^^ expected identifier, found keyword

error: expected identifier, found keyword `use`
  --> $DIR/bound-modifiers.rs:11:38
   |
LL | fn asyncness() -> impl Sized + async use<> {}
   |                                      ^^^ expected identifier, found keyword

error: expected identifier, found keyword `use`
  --> $DIR/bound-modifiers.rs:16:38
   |
LL | fn constness() -> impl Sized + const use<> {}
   |                                      ^^^ expected identifier, found keyword

error: expected identifier, found keyword `use`
  --> $DIR/bound-modifiers.rs:21:37
   |
LL | fn binder() -> impl Sized + for<'a> use<> {}
   |                                     ^^^ expected identifier, found keyword

error[E0405]: cannot find trait `r#use` in this scope
  --> $DIR/bound-modifiers.rs:5:32
   |
LL | fn polarity() -> impl Sized + ?use<> {}
   |                                ^^^ not found in this scope

error[E0405]: cannot find trait `r#use` in this scope
  --> $DIR/bound-modifiers.rs:11:38
   |
LL | fn asyncness() -> impl Sized + async use<> {}
   |                                      ^^^ not found in this scope

error[E0405]: cannot find trait `r#use` in this scope
  --> $DIR/bound-modifiers.rs:16:38
   |
LL | fn constness() -> impl Sized + const use<> {}
   |                                      ^^^ not found in this scope

error[E0405]: cannot find trait `r#use` in this scope
  --> $DIR/bound-modifiers.rs:21:37
   |
LL | fn binder() -> impl Sized + for<'a> use<> {}
   |                                     ^^^ not found in this scope

error[E0658]: async closures are unstable
  --> $DIR/bound-modifiers.rs:11:32
   |
LL | fn asyncness() -> impl Sized + async use<> {}
   |                                ^^^^^
   |
   = note: see issue #62290 <https://github.com/rust-lang/rust/issues/62290> for more information
   = help: add `#![feature(async_closure)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = help: to use an async block, remove the `||`: `async {`

error[E0658]: const trait impls are experimental
  --> $DIR/bound-modifiers.rs:16:32
   |
LL | fn constness() -> impl Sized + const use<> {}
   |                                ^^^^^
   |
   = note: see issue #67792 <https://github.com/rust-lang/rust/issues/67792> for more information
   = help: add `#![feature(const_trait_impl)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

warning: relaxing a default bound only does something for `?Sized`; all other traits are not bound by default
  --> $DIR/bound-modifiers.rs:5:31
   |
LL | fn polarity() -> impl Sized + ?use<> {}
   |                               ^^^^^^

warning: relaxing a default bound only does something for `?Sized`; all other traits are not bound by default
  --> $DIR/bound-modifiers.rs:5:31
   |
LL | fn polarity() -> impl Sized + ?use<> {}
   |                               ^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: aborting due to 10 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0405, E0658.
For more information about an error, try `rustc --explain E0405`.
